### PR TITLE
Revised Material Gallery demo

### PR DESCRIPTION
**Pull request by HansMuller**
The demo's home page is now a grid divided into the same six sections that appear in the Material Design spec itself.  Tapping on a section leads to a list of demos. Currently only the "Components" section has any demos.

All of the demos are now self-contained apps with their own Scaffold and etc.

### CODE REVIEW

**Comment by eseidelGoogle** [comment]
Hard for me to visualize this change.

**Comment by HansMuller** [comment]
I posted the unfinished CL because I'd finished a pass at the structural changes. They're somewhat complicated by the need to manage the routeNameToBuilder Map so that the demo list items can be launched by pushing a named route and the demos themselves can be specified with a WidgetBuilder.

At any rate. Here's a screenshot of the current home screen. Clicking on a demo causes it to slide in from the left.

screen

File: examples/material_gallery/lib/demo/drop_down_demo.dart [hunk 1]
```diff
@@ -4,16 +4,14 @@
...
   _DropDownDemoState createState() => new _DropDownDemoState();
 }
 
 class _DropDownDemoState extends State<DropDownDemo> {
-  String _value = "Free";
+  String value = "Free";
 
-  List<DropDownMenuItem<String>> _buildItems() {
+  List<DropDownMenuItem<String>> buildItems() {
     return ["One", "Two", "Free", "Four"].map((String value) {
```
Comment (Hixie) [inline examples/material_gallery/lib/demo/drop_down_demo.dart]: `<String>[...]`

**Comment by Hixie** [comment]
(squash your commits before checkin)

File: examples/material_gallery/lib/demo/drop_down_demo.dart (hunk 1 above)
Comment (Hixie) [inline examples/material_gallery/lib/demo/drop_down_demo.dart]: missing

**Comment by Hixie** [comment]
lgtm, but travis is angry. missing file?

**Comment by HansMuller** [comment]
Yes: I had removed the progress_indicator widget example, and created a gallery demo version. There was collision with Adam's animation changes. I need to restore order :-).

**Pull request merged**